src/RageSoundReader.cpp

```diff
@@ -6,34 +6,34 @@
 REGISTER_CLASS_TRAITS( RageSoundReader, pCopy->Copy() );
 
 /* Read(), handling the STREAM_LOOPED and empty return cases. */
-int RageSoundReader::RetriedRead( float *pBuffer, int iFrames, int *iSourceFrame, float *fRate )
-{
-	/* pReader may return 0, which means "try again immediately".  As a failsafe,
-	 * only try this a finite number of times.  Use a high number, because in
-	 * principle each filter in the stack may cause this. */
-	int iTries = 10;
-	while( --iTries )
-	{
-		if( fRate )
+int RageSoundReader::RetriedRead(float* pBuffer, int iFrames, int* iSourceFrame, float* fRate) {
+	if (iFrames == 0) {
+		return 0;
+	}
+
+	// pReader may return 0, which means "try again immediately". As a failsafe,
+	// only try this a finite number of times. Use a high number, because in
+	// principle each filter in the stack may cause this.
+	std::uint8_t retryCount = 50;
```

Comment:
```
#include <cstdint>
```
@sukibaby

Reply:
That does work but it seems like users on Linux in the build test thread were seeing mixed behavior even with cstdint so I replaced the `uint8_t` with an `unsigned` in #464 .